scripts/configure_pyproject_toml.py

```diff
@@ -122,6 +122,20 @@ def parse_args():
         "scipy-openblas32>=0.3.26",
     ]
 
+    if backend == "lightning_gpu":
+        dependencies += ["custatevec-cu12"]
+
+    if backend == "lightning_tensor":
+        dependencies += ["cutensornet-cu12", "cutensor-cu12>=2.0.2", "nvidia-cusolver-cu12"]
```

Comment:
Do we have to update `requirements.txt`?

Reply:
Perhaps we should add them to `requirements-dev.txt`? 

I am also not sure whether e.g. the cuda runtime libraries should be added; can we assume that if people are compiling LGPU from source, they will have the appropriate cuda-runtime libraries, e.g. from nvidia-toolkit? Or is it better to have them all here so we can make sure their versions match? I am also wondering if it's a good idea to have a `requirements-gpu.txt` separately from `requirements-dev.txt`, so that people who don't want to compile specifically for LGPU (e.g. if they just want LQ/LK) don't need to pull any of the nvidia libraries. What do we think? cc@maliasadi

Reply:
So from discussions, `requirements-dev.txt` is mainly for people installing from source, and for people building for LGPU, the lower level libraries (e.g. cublas/nvjitlink) should already be available from the cuda-toolkit, which will be a better source to install them, rather than from our requirements.txt.

Reply:
So I won't include them here in the `requirements.txt` or `requirements-dev.txt`. 

However, something else I will do is to make `custatevec-cu12` and `cutensornet-cu12` in our `requirements-dev.txt` dependent on Linux platforms only (e.g. not build on mac). This is because they're only built for linux, and cannot be pip installed on macs.

Reply:
> However, something else I will do is to make custatevec-cu12 and cutensornet-cu12 in our requirements-dev.txt dependent on Linux platforms only (e.g. not build on mac). This is because they're only built for linux, and cannot be pip installed on macs.

Good call! Thanks @josephleekl .